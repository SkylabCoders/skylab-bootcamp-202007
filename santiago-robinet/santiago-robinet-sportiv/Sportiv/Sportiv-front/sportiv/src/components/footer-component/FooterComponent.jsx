import React from "react";
import "./FooterComponent.scss";
import { NavLink } from 'react-router-dom';
import { useAuth0 } from '@auth0/auth0-react';


function Footer() {

  const { isAuthenticated, loginWithRedirect, logoutWithRedirect, user } = useAuth0();

  return (
    <footer className="footer__main">
      <div className="footer__left">
        <div className="footer-other-pages footer-icon">SANTIAGOROBINET</div>
        {/* <div className="footer-search footer-icon"><NavLink to="/search"><img src="https://image.flaticon.com/icons/svg/751/751463.svg" alt="hamburguer-button"/></NavLink></div> */}
      </div>
      <div className="footer-logo__center">
        <div className="footer-logo"><NavLink to="/">LOGO</NavLink></div>
      </div>
      <div className="footer__right">
        <div className="footer-explore footer-icon"><NavLink to="" ><img src="https://image.flaticon.com/icons/svg/684/684908.svg" alt="explore-button"/></NavLink></div>
        {isAuthenticated && 
          <div className="footer-user footer-icon"><NavLink to="/profile"><img className="footer-profile" src={user.picture} alt="profile-button"/></NavLink></div>
        }
        {!isAuthenticated && 
          <div className="footer-user footer-icon"><NavLink to="" onClick={() => loginWithRedirect()}><img src="https://image.flaticon.com/icons/svg/2170/2170153.svg" alt="login-button"/></NavLink></div>
        }

      </div>
    </footer>
  );
}

export default Footer;
