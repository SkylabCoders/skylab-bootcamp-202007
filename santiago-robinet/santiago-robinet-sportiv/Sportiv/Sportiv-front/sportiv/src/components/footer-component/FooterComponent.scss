

.footer__main{
    display: none;
    flex-direction: row;
    background-color: rgba(0, 0, 0, 0.719);
    height: 60px;
    width: 100%;
    margin-bottom: 0;
   
}

.footer__left{
    display: flex;
    flex-direction: row;
    width: 33%;
    justify-content: space-around;
    align-items: center;
}

.footer-logo__center{
    display: flex;
    flex-direction: row;
    width: 33%;
    justify-content: center;
    align-items: center;
}

.footer__right{
    display: flex;
    flex-direction: row;
    width: 33%;
    justify-content: space-around;
    align-items: center;
}

.footer-icon{
    width: 30px;
    height: 30px;
    
}

.footer-icon  img{
    width: 100%;
}

.footer-profile{
    border-radius: 50%;
}

@media (min-width: 750px){
    .footer__main{
        display: flex;

    }
}