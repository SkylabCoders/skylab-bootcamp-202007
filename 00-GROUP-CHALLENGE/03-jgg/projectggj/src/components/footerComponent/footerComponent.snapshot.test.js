import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import FooterComponent from './footerComponent';


describe('footerComponent', () => {
    const footerComponentTree = renderer.create(
        <Router>
            <FooterComponent />
        </Router>
    );
    it('should macth', () => {
        expect(footerComponentTree).toMatchSnapshot();
    })
});