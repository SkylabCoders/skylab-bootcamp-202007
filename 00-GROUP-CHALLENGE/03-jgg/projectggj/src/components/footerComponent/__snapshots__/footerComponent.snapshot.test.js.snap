// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footerComponent should macth 1`] = `
<div
  className="mainContainer"
>
  <img
    alt="Dragon Ball One Start"
    onClick={[Function]}
    src="https://i.pinimg.com/originals/a5/f9/a2/a5f9a2eb5c0bfb1f66988696e1f31334.png"
    title="Go Back"
  />
  <p>
    By Gemma Sanz | Gabriel Penalva | Jordi Mu√±oz
  </p>
</div>
`;
